pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,220 @@
+package mustgather
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	mustGatherLong = templates.LongDesc(`
+		Launch a pod to gather debugging information
+
+		This command will launch a pod in a temporary namespace on your
+		cluster that gathers debugging information, using a copy of the active 
+		client config context, and then downloads the gathered information. 
+	`)
+
+	mustGatherExample = templates.Examples(`
+		# gather default information using the default image and command, writing into ./must-gather.local.<rand>
+		  oc adm must-gather
+
+		# gather default information with a specific local folder to copy to
+		  oc adm must-gather --dest-dir=/local/directory
+
+		# gather default information using a specific image, command, and pod-dir
+		  oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
+	`)
+)
+
+type MustGatherFlags struct {
+	ConfigFlags *genericclioptions.ConfigFlags
```

Comment:
You should not need this, since you're part of `oc` this is implicit.

Reply:
No longer applicable.